src/hooks/use-render.js

```diff
@@ -1,12 +1,17 @@
 import { useEffect } from '@nebula.js/stardust';
 
-const useRender = ({ settings, models }) => {
+const useRender = ({ settings, models, chart }) => {
   useEffect(() => {
     if (!settings) {
       return;
     }
 
-    const { chartModel } = models;
+    const { chartModel, dockService, layoutService } = models;
+    const { width, height } = dockService.meta.chart.size;
+    const { clientWidth: w, clientHeight: h } = chart?.element;
```

Comment:
what happen if chart is undefined? should we do something like the following if chart is undefined
const { clientWidth: w, clientHeight: h } = chart?.element || {};

Reply:
You are right. I will return immediately if chart is undefined. Because in that case, we cannot call chartModel.command.update, which will call chart.update.